### DISCUSSION TITLE
OCPP services documentation: updating max charge current

### DISCUSSION THREAD

**Discussion by Mougli** [lbbrhzn/ocpp#1185] [Q&A]
Hello,

I've made a sensor that calculates the max charge current based on my excess solar panel production, and I'm planning on using it in a periodic automation to make sure that the EV only charges from solar and doesn't eat into the home battery.

Home Assistant's default Set Number service works to change the charge current... as long as I feed it a static value, can't seem to be able to use templates in there.

I thought I'd use the service OCCP Set Maximum Charge Rate, hoping the service data would allow templates, but I don't know what the correct data structure looks like, and I didn't find it in the documentation.

Does anyone know the correct syntax, or whether this is possible in the first place?

Thanks!

**Comment by archerjm** [comment]
I do something similar using four automations (using a conservative 240V value for voltage for power conversions to current)
1. based on grid export being above 240 W.  Set charge rate to export power (positive in my case) divided by 240V to get current plus existing charge current.  The action is:
`service: number.set_value
data:
  value: >
    {%if ((states.number.charger_maximum_current.state | int ) + (((
    states.sensor.grid_power.state | float ) / 240) | int) ) > 32 %} 
      {{ 32 }} 
    {% else %} 
      {{ ((states.number.charger_maximum_current.state | int ) + ((( states.sensor.grid_power.state | float ) / 240) | int)) }}
    {% endif %}
target:
  entity_id: number.charger_maximum_current
`
2. Based on house battery charge rate being below 240 W.  The action for this one is:
`service: number.set_value
data:
  value: >
    {%if ((states.number.charger_maximum_current.state | int ) + (((
    states.sensor.battery_charge_rate.state | float ) / 240) | int) ) < 6 %} 
      {{ 6 }} 
    {% else %} 
      {{ ((states.number.charger_maximum_current.state | int ) + ((( states.sensor.battery_charge_rate.state | float ) / 240) | int)) }}
    {% endif %}
target:
  entity_id: number.charger_maximum_current
`
3. A simple automation to start charging based on grid feed in power being higher than a set value.
4. A trigger to stop the charger based on house battery discharge rate grid.

Hopefully this helps.

**Reply by Mougli** [reply]
That's helpful, thanks! I did not realise that I could go in YAML mode in the services tab of Developer Tools to see the available parameters for a service.

I've managed to do what I want with the Number.SetValue service, but out of curiosity I tried the corresponding OCPP service, and service call succeeds with either limit_amps or limit_watts parameters, but nothing actually seems to happen? I'm expecting that to change the number.charger_maximum_current (if only indirectly), but that's not the case. Is it supposed to work?

(I have a Simpson & Partners charger)

**Comment by blacknell** [comment]
The best way to do this is now [documented](https://github.com/lbbrhzn/ocpp/blob/main/docs/Charge_automation.md) in the most recent release.

Relates #1496
